lib/sentry/envelope.ex

```diff
@@ -34,6 +34,17 @@ defmodule Sentry.Envelope do
     }
   end
 
+  @doc """
+  Creates a new envelope containing the client report and a map of discarded events.
+  """
+  @spec from_client_report(ClientReport.t()) :: t()
+  def from_client_report(%ClientReport{} = client_report) do
```

Comment:
**Blocking**: we need a test for this function and a test for `to_binary/1` with a client report as arg.

Reply:
Still need to add this test and others. I'll add more test after the logic stops moving so maybe one more round of feedback from @whatyouhide

Reply:
This part of the logic won't change because encoding is simple enough here, so you can go ahead and add tests for this anytime